### ISSUE TITLE
Deserialization into simple char causes System.IndexOutOfRangeException

### ISSUE THREAD

**Issue by Micke90s** [rpgmaker/NetJSON#227]
Hallo,

could you please check the following behavior?

I tried to update NetJSON from 1.2.1.8 (I am not absolute sure about the version. It also could be older) to the current version. Unfortunately I got the problem that the existing classes causing the exception “System.IndexOutOfRangeException” when char is used as attribute.

```C#
  public class CNTL_SIMPLE_EVENT {
    public int EVENTID;
    //public String APPL_EVENT; //Works perfect
    public char APPL_EVENT; //Causes Exception
  }
  class Program  {
    static void Main(string[] args)    {
      var json = "{\"EVENTID\":19 ,\"APPL_EVENT\":\"\"}";
      var simple = NetJSON.NetJSON.Deserialize<CNTL_SIMPLE_EVENT>(json);
    }
  }
```

**Comment by rpgmaker** [comment]
Sure, will take a look at it.

Thanks,

**Comment by rpgmaker** [comment]
Please test again. I have fixed and published it to nuget as 1.3.3 which should be available soon. 

Thanks again for the finding.

**Comment by Micke90s** [comment]
Thank you very much. Now it works perfect.

--------------------

At the moment I have a second problem with the update from 1.2.1.8 to 1.3.2/1.3.3. I think that the problem is caused by my project configuration. Unfortunately, I haven't been able to find the problem yet. If you would have any tip for me it would be great. 

After the update the following exception is thrown:
```
System.TypeInitializationException: "The type initializer for "NetJSONCachedSerializer`1" threw an exception."
Inner Exception
System.TypeLoadException: Access is denied: "NetJSON.NetJSONSerializer`1[System.Collections.Generic.List`1[bit.CNTL_SIMPLE_EVENT]]".
```
![Exception](https://user-images.githubusercontent.com/59920485/72716939-65000780-3b73-11ea-880c-7a0e3d2e8e35.jpg)

The direct access works fine:
```C#
        //NetJson.cs
        internal static NetJSONSerializer<T> GetSerializer<T>() {
            //Works fine
            List<bit.CNTL_SIMPLE_EVENT> list = new List<bit.CNTL_SIMPLE_EVENT>();
            //Thorws exception
            return NetJSONCachedSerializer<T>.Serializer;
        }
```

The used class is defined as public `public class CNTL_SIMPLE_EVENT` and part of the same project as the NetJSON cs-file

Before update:
![CurrentVersion](https://user-images.githubusercontent.com/59920485/72716865-41d55800-3b73-11ea-9b86-5c04c0ac79c2.jpg)

After update:
![NewVersion](https://user-images.githubusercontent.com/59920485/72716917-59144580-3b73-11ea-806a-62482da12f86.jpg)

It would be great if you could give me a tip to solve this.

Thank you
